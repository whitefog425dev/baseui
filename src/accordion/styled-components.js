/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import {styled} from '../styles/index.js';
import {getSvgStyles} from '../icon/styled-components.js';
import type {SharedStylePropsT} from './types.js';

/**
 * Main component container element
 */
export const Root = styled('div', (props: SharedStylePropsT) => {
  return {};
});

export const PanelContainer = styled('div', (props: SharedStylePropsT) => {
  return {width: '100%'};
});

export const Header = styled('h3', (props: SharedStylePropsT) => {
  const {
    $disabled,
    $expanded,
    $theme: {colors, sizing, typography},
  } = props;
  return {
    ...typography.font450,
    color: colors.foreground,
    cursor: $disabled ? 'not-allowed' : 'pointer',
    background: colors.listHeaderFill,
    paddingTop: sizing.scale600,
    paddingBottom: sizing.scale600,
    paddingLeft: sizing.scale700,
    paddingRight: sizing.scale1200,
    marginTop: 0,
    marginBottom: 0,
    marginLeft: 0,
    marginRight: 0,
    borderBottom: `1px solid ${$expanded ? colors.mono500 : colors.mono400}`,
    position: 'relative',
    display: 'flex',
    alignItems: 'center',
    ':hover': {
      color: colors.linkHover,
    },
    ':focus': {
      backgroudColor: colors.background,
    },
  };
});

export const ToggleIcon = styled('svg', props => {
  const {$theme, $disabled, $color} = props;
  return {
    ...getSvgStyles(props),
    color: $color || $theme.colors.foreground,
    cursor: $disabled ? 'not-allowed' : 'pointer',
    position: 'absolute',
    right: '12px',
  };
});

export const Content = styled('div', (props: SharedStylePropsT) => {
  const {
    $theme: {animation, colors, sizing, typography},
    $expanded,
  } = props;
  return {
    ...typography.font300,
    backgroundColor: colors.background,
    color: colors.foreground,
    paddingTop: $expanded ? sizing.scale800 : 0,
    paddingBottom: $expanded ? sizing.scale1000 : 0,
    paddingLeft: sizing.scale800,
    paddingRight: sizing.scale800,
    marginTop: 0,
    marginBottom: 0,
    marginLeft: 0,
    marginRight: 0,
    borderBottom: $expanded ? `1px solid ${colors.border}` : 'none',
    boxSizing: 'border-box',
    height: $expanded ? 'auto' : '0',
    maxHeight: $expanded ? '100%' : '0',
    overflow: 'hidden',
    opacity: $expanded ? 1 : 0,
    transitionProperty: 'all',
    transitionDuration: animation.timing400,
    transitionTimingFunction: animation.easeInOutCurve,
  };
});
