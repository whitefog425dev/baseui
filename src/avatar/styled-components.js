/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

import {styled} from '../styles/index.js';
import type {StylePropsT} from './types.js';

function getSize(props: StylePropsT) {
  const {$size, $theme} = props;

  const defaultSize = $theme.sizing.scale1000;
  const size = $size || defaultSize;
  return $theme.sizing[size] || size;
}

export const Avatar = styled('img', (props: StylePropsT) => {
  const themedSize = getSize(props);

  return {
    borderRadius: '50%',
    boxSizing: 'border-box',
    display: 'block',
    height: themedSize,
    width: themedSize,
  };
});
Avatar.displayName = 'StyledAvatar';

export const Root = styled('div', (props: StylePropsT) => {
  const {$didImageFailToLoad} = props;
  const themedSize = getSize(props);

  return {
    backgroundColor: props.$theme.colors.mono400,
    borderRadius: '50%',
    boxSizing: 'border-box',
    display: 'inline-block',

    // image previously set the root height/width
    // since image is not rendered, set the height/width
    height: $didImageFailToLoad ? themedSize : null,
    width: $didImageFailToLoad ? themedSize : null,
  };
});
Root.displayName = 'StyledRoot';
