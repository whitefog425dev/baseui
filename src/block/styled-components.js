/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

// @flow

import {styled} from '../styles/index.js';
import type {Font} from '../styles/types.js';
import type {StyledBlockPropsT} from './types.js';

export const StyledBlock = styled(
  'div',
  ({
    $theme,
    $color,
    $font,
    $alignContent,
    $alignItems,
    $alignSelf,
    $flexDirection,
    $display,
    $flex,
    $justifyContent,
    $justifySelf,
    $position,
    $width,
    $minWidth,
    $maxWidth,
    $height,
    $minHeight,
    $maxHeight,
    $overflow,
    $margin,
    $marginTop,
    $marginRight,
    $marginBottom,
    $marginLeft,
    $padding,
    $paddingTop,
    $paddingRight,
    $paddingBottom,
    $paddingLeft,
    $flexWrap,
    $left,
    $top,
    $right,
    $bottom,
    ...other
  }: StyledBlockPropsT) => {
    const {colors, typography, sizing} = $theme;

    let styles = {};

    if ($color !== undefined) {
      styles.color = colors[$color] || $color;
    }

    if ($font !== undefined) {
      const font: Font = typography[$font];
      styles.fontFamily = font.fontFamily;
      styles.fontWeight = font.fontWeight;
      styles.fontSize = font.fontSize;
      styles.lineHeight = font.lineHeight;
    }

    if ($alignContent !== undefined) {
      styles.alignContent = $alignContent;
    }

    if ($alignItems !== undefined) {
      styles.alignItems = $alignItems;
    }

    if ($alignSelf !== undefined) {
      styles.alignSelf = $alignSelf;
    }

    if ($flexDirection !== undefined) {
      styles.flexDirection = $flexDirection;
    }

    if ($display !== undefined) {
      styles.display = $display;
    }

    if ($flex !== undefined) {
      styles.flex = $flex;
    }

    if ($justifyContent !== undefined) {
      styles.justifyContent = $justifyContent;
    }

    if ($justifySelf !== undefined) {
      styles.justifySelf = $justifySelf;
    }

    if ($position !== undefined) {
      styles.position = $position;
    }

    if ($width !== undefined) {
      styles.width = $width;
    }

    if ($minWidth !== undefined) {
      styles.minWidth = $minWidth;
    }

    if ($maxWidth !== undefined) {
      styles.maxWidth = $maxWidth;
    }

    if ($height !== undefined) {
      styles.height = $height;
    }

    if ($minHeight !== undefined) {
      styles.minHeight = $minHeight;
    }

    if ($maxHeight !== undefined) {
      styles.maxHeight = $maxHeight;
    }

    if ($overflow === 'scrollX') {
      styles.overflowX = 'scroll';
    } else if ($overflow === 'scrollY') {
      styles.overflowY = 'scroll';
    } else if ($overflow !== undefined) {
      styles.overflow = $overflow;
    }

    if ($margin !== undefined) {
      styles.margin = sizing[$margin] || $margin;
    }

    if ($marginTop !== undefined) {
      styles.marginTop = sizing[$marginTop] || $marginTop;
    }

    if ($marginRight !== undefined) {
      styles.marginRight = sizing[$marginRight] || $marginRight;
    }

    if ($marginBottom !== undefined) {
      styles.marginBottom = sizing[$marginBottom] || $marginBottom;
    }

    if ($marginLeft !== undefined) {
      styles.marginLeft = sizing[$marginLeft] || $marginLeft;
    }

    if ($padding !== undefined) {
      styles.padding = sizing[$padding] || $padding;
    }

    if ($paddingTop !== undefined) {
      styles.paddingTop = sizing[$paddingTop] || $paddingTop;
    }

    if ($paddingRight !== undefined) {
      styles.paddingRight = sizing[$paddingRight] || $paddingRight;
    }

    if ($paddingBottom !== undefined) {
      styles.paddingBottom = sizing[$paddingBottom] || $paddingBottom;
    }

    if ($paddingLeft !== undefined) {
      styles.paddingLeft = sizing[$paddingLeft] || $paddingLeft;
    }

    if ($flexWrap !== undefined) {
      styles.flexWrap = 'wrap';
    }

    if ($left !== undefined) {
      styles.left = $left;
    }

    if ($top !== undefined) {
      styles.top = $top;
    }

    if ($right !== undefined) {
      styles.right = $right;
    }

    if ($bottom !== undefined) {
      styles.bottom = $bottom;
    }

    return styles;
  },
);
