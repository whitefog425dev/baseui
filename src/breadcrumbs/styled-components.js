/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

// @flow

import {styled} from '../styles/index.js';
import type {StyledRootPropsT, StyledSeparatorT} from './types.js';
import {ChevronRight} from '../icon/index.js';

export const StyledRoot = styled('nav', ({$theme}: StyledRootPropsT) => {
  return {
    color: $theme.colors.breadcrumbsText,
    ...$theme.typography.font450,
  };
});
StyledRoot.displayName = 'StyledRoot';

export const StyledSeparator = styled('div', ({$theme}: StyledSeparatorT) => {
  return {
    display: 'inline-block',
    color: $theme.colors.breadcrumbsSeparatorFill,
    marginLeft: $theme.sizing.scale300,
    marginRight: $theme.sizing.scale300,
  };
});
StyledSeparator.displayName = 'StyledSeparator';

export const StyledIcon = styled(ChevronRight, () => {
  return {
    verticalAlign: 'text-bottom',
  };
});
StyledIcon.displayName = 'StyledIcon';
