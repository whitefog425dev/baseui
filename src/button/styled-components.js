/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import {styled} from '../styles/index.js';
import {KIND, SIZE, SHAPE} from './constants.js';
import type {ThemeT} from '../styles/index.js';
import type {SharedStylePropsT} from './types.js';

type StylePropsT = SharedStylePropsT & {
  $theme: ThemeT,
};

export const BaseButton = styled(
  'button',
  ({$theme, $size, $kind, $shape, $isLoading}: StylePropsT) => ({
    position: 'relative',
    ...($size === SIZE.compact
      ? $theme.typography.font250
      : $theme.typography.font350),
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'center',
    border: 'none',
    borderRadius:
      $shape === SHAPE.round
        ? '50%'
        : $theme.borders.useRoundedCorners
          ? $theme.borders.radius200
          : '0px',
    textDecoration: 'none',
    outline: 'none',
    WebkitAppearance: 'none',
    transitionProperty: 'background',
    transitionDuration: $theme.animation.timing100,
    transitionTimingFunction: $theme.animation.easeOutCurve,
    cursor: 'pointer',
    ':disabled': {
      cursor: 'not-allowed',
      backgroundColor: $theme.colors.buttonDisabledFill,
      color: $theme.colors.buttonDisabledText,
    },
    // Padding For Shape and Size
    ...getStyleForShape({$theme, $shape, $size}),
    // Kind style override
    ...getStyleForKind({$theme, $kind, $isLoading}),
  }),
);
BaseButton.displayName = 'StyledBaseButton';

export const EndEnhancer = styled('div', ({$theme}: StylePropsT) => ({
  display: 'flex',
  marginLeft: $theme.sizing.scale500,
}));
EndEnhancer.displayName = 'StyledEndEnhancer';

export const StartEnhancer = styled('div', ({$theme}: StylePropsT) => ({
  display: 'flex',
  marginRight: $theme.sizing.scale500,
}));
StartEnhancer.displayName = 'StyledStartEnhancer';

export const LoadingSpinnerContainer = styled('div', {
  // To center within parent
  position: 'absolute',
  left: '50%',
  top: '50%',
  transform: 'translate(-50%, -50%)',
});
LoadingSpinnerContainer.displayName = 'StyledLoadingSpinnerContainer';

export const LoadingSpinner = styled(
  'div',
  ({$theme, $kind, $disabled}: StylePropsT) => {
    const {foreground, background} = getLoadingSpinnerColors({
      $theme,
      $kind,
      $disabled,
    });
    return {
      height: $theme.sizing.scale600,
      width: $theme.sizing.scale600,
      borderRadius: '50%',
      borderStyle: 'solid',
      borderWidth: $theme.sizing.scale0,
      borderTopColor: foreground,
      borderLeftColor: background,
      borderBottomColor: background,
      borderRightColor: background,
      animationDuration: $theme.animation.timing700,
      animationTimingFunction: 'linear',
      animationIterationCount: 'infinite',
      animationName: {
        to: {
          transform: 'rotate(360deg)',
        },
        from: {
          transform: 'rotate(0deg)',
        },
      },
    };
  },
);
LoadingSpinner.displayName = 'StyledLoadingSpinner';

export function getLoadingSpinnerColors({
  $theme,
  $kind,
  $disabled,
}: StylePropsT) {
  return {
    foreground: $disabled
      ? $theme.colors.mono600
      : $kind === KIND.primary
        ? $theme.colors.white
        : $theme.colors.primary,
    background: $disabled
      ? 'rgba(179, 179, 179, 0.32)'
      : $kind === KIND.primary
        ? 'rgba(255, 255, 255, 0.32)'
        : 'rgba(39, 110, 241, 0.32)',
  };
}

export function getStyleForShape({$theme, $shape, $size}: StylePropsT) {
  switch ($shape) {
    case SHAPE.round:
    case SHAPE.square:
      return {
        paddingTop:
          $size === SIZE.compact
            ? $theme.sizing.scale400
            : $theme.sizing.scale500,
        paddingBottom:
          $size === SIZE.compact
            ? $theme.sizing.scale400
            : $theme.sizing.scale500,
        paddingLeft:
          $size === SIZE.compact
            ? $theme.sizing.scale400
            : $theme.sizing.scale500,
        paddingRight:
          $size === SIZE.compact
            ? $theme.sizing.scale400
            : $theme.sizing.scale500,
      };
    default:
      return {
        paddingTop:
          $size === SIZE.compact
            ? $theme.sizing.scale200
            : $theme.sizing.scale300,
        paddingBottom:
          $size === SIZE.compact
            ? $theme.sizing.scale200
            : $theme.sizing.scale300,
        paddingLeft: $theme.sizing.scale600,
        paddingRight: $theme.sizing.scale600,
      };
  }
}

export function getStyleForKind({$theme, $kind, $isLoading}: StylePropsT) {
  switch ($kind) {
    case KIND.primary:
      return {
        color: $theme.colors.buttonPrimaryText,
        backgroundColor: $theme.colors.buttonPrimaryFill,
        ':hover:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.buttonPrimaryActive
            : $theme.colors.buttonPrimaryHover,
        },
        ':focus:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.buttonPrimaryActive
            : $theme.colors.buttonPrimaryHover,
        },
        ':active:enabled': {
          backgroundColor: $theme.colors.buttonPrimaryActive,
        },
      };
    case KIND.secondary:
      return {
        color: $theme.colors.buttonSecondaryText,
        backgroundColor: $theme.colors.buttonSecondaryFill,
        ':hover:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.buttonSecondaryActive
            : $theme.colors.buttonSecondaryHover,
        },
        ':focus:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.buttonSecondaryActive
            : $theme.colors.buttonSecondaryHover,
        },
        ':active:enabled': {
          backgroundColor: $theme.colors.buttonSecondaryActive,
        },
      };
    case KIND.tertiary:
      return {
        color: $theme.colors.buttonTertiaryText,
        backgroundColor: $theme.colors.buttonTertiaryFill,
        ':hover:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.buttonTertiaryActive
            : $theme.colors.buttonTertiaryHover,
        },
        ':focus:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.buttonTertiaryActive
            : $theme.colors.buttonTertiaryHover,
        },
        ':active:enabled': {
          backgroundColor: $theme.colors.buttonTertiaryActive,
        },
      };
    case KIND.minimal:
      return {
        color: $theme.colors.buttonMinimalText,
        backgroundColor: $theme.colors.buttonMinimalFill,
        ':hover:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.buttonMinimalActive
            : $theme.colors.buttonMinimalHover,
        },
        ':focus:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.buttonMinimalActive
            : $theme.colors.buttonMinimalHover,
        },
        ':active:enabled': {
          backgroundColor: $theme.colors.buttonMinimalActive,
        },
      };
    default:
      return {};
  }
}
