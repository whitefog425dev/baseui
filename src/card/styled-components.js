/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

/* @flow */

import {styled} from '../styles/index.js';

export const Action = styled('div', () => ({}));
Action.displayName = 'StyledAction';

export const Body = styled('div', ({$theme}) => {
  const {colors, sizing, typography} = $theme;
  return {
    marginBottom: sizing.scale600,
    color: colors.foregroundAlt,
    ...typography.font400,
  };
});
Body.displayName = 'StyledBody';

export const Contents = styled('div', ({$theme}) => {
  const {sizing} = $theme;
  return {
    margin: sizing.scale800,
  };
});
Contents.displayName = 'StyledContents';

export const HeaderImage = styled('img', ({$theme}) => {
  const {borders} = $theme;
  return {
    borderTopLeftRadius: borders.radius200,
    borderTopRightRadius: borders.radius200,
    objectFit: 'contain',
    maxWidth: '100%',
  };
});
HeaderImage.displayName = 'StyledHeaderImage';

export const Root = styled('div', ({$theme}) => {
  const {borders, lighting, colors} = $theme;
  return {
    ...borders.border300,
    boxShadow: lighting.shadow400,
    borderRadius: borders.radius200,
    backgroundColor: colors.backgroundAlt,
  };
});
Root.displayName = 'StyledWrapper';

export const Thumbnail = styled('img', props => {
  const {
    $theme: {borders, sizing},
  } = props;
  return {
    float: 'right',
    height: sizing.scale2400,
    width: sizing.scale2400,
    objectFit: 'cover',
    borderRadius: borders.radius200,
    ...borders.border200,
    margin: `0 0 ${sizing.scale500} ${sizing.scale500}`,
  };
});
Thumbnail.displayName = 'StyledThumbnail';

export const Title = styled('h1', ({$hasThumbnail, $theme}) => {
  const {colors, sizing, typography} = $theme;
  return {
    ...typography.font500,
    color: colors.foreground,
    fontWeight: 500,
    margin: `0 0 ${sizing.scale300} 0`,
    padding: 0,
  };
});
Title.displayName = 'StyledTitle';
