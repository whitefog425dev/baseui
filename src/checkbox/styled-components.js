// @flow
import {styled} from '../styles';

function getBorderColor(props) {
  const {$checked, $isError, $isIndeterminate, $theme} = props;
  const {colors} = $theme;
  return $isError
    ? colors.negative400
    : $isIndeterminate || $checked ? colors.primary400 : colors.mono700;
}

function getLabelPadding(props) {
  const {$labelPlacement = '', $theme} = props;
  const {sizing} = $theme;
  const {scale200} = sizing;
  const paddingSide =
    'padding' + $labelPlacement.replace(/^\w/, c => c.toUpperCase());
  return {
    [paddingSide]: scale200,
  };
}

function getBackgroundColor(props) {
  const {
    $disabled,
    $checked,
    $isIndeterminate,
    $isFocused,
    $isError,
    $isHovered,
    $theme,
  } = props;
  const {colors} = $theme;
  return $disabled
    ? colors.mono300
    : $isError
      ? colors.negative400
      : $isIndeterminate || $checked
        ? colors.primary400
        : $isFocused ? colors.mono500 : $isHovered ? colors.mono400 : null;
}

function getLabelColor(props) {
  const {$disabled, $theme} = props;
  const {colors} = $theme;
  return $disabled ? colors.mono600 : colors.mono1000;
}

export const Root = styled('label', props => {
  const {$disabled, $labelPlacement} = props;
  return {
    flexDirection:
      $labelPlacement === 'top' || $labelPlacement === 'bottom'
        ? 'column'
        : 'row',
    display: 'flex',
    alignItems: 'center',
    cursor: $disabled ? 'not-allowed' : 'pointer',
  };
});

export const Checkmark = styled('span', props => {
  const {
    $checked,
    $disabled,
    $isIndeterminate,
    $theme,
    $isFocused,
    $isError,
  } = props;
  const {colors, sizing, animation} = $theme;
  const activeStyle = {
    backgroundColor:
      $checked || $isIndeterminate || $isError
        ? null
        : $isFocused
          ? colors.mono500
          : !$disabled && !$isIndeterminate && !$checked
            ? colors.mono400
            : null,
  };
  return {
    flex: '0 0 auto',
    transitionDuration: animation.timing100,
    transitionTimingFunction: animation.easeOutCurve,
    width: sizing.scale600,
    height: sizing.scale600,
    left: '4px',
    top: '4px',
    borderStyle: 'solid',
    borderWidth: '2px',
    borderColor: getBorderColor(props),
    borderRadius: $theme.borders.useRoundedCorners
      ? $theme.borders.radius200
      : null,
    display: 'inline-block',
    verticalAlign: 'middle',
    backgroundImage: $isIndeterminate
      ? 'url(\'data:image/svg+xml;utf8,<svg width="12" height="2" viewBox="0 0 12 2" fill="none" xmlns="http://www.w3.org/2000/svg"><line x1="1" y1="-1" x2="11" y2="-1" transform="translate(0 2)" stroke="white" stroke-width="2" stroke-linecap="round"/></svg>\');'
      : $checked
        ? 'url(\'data:image/svg+xml;utf8,<svg width="11" height="10" viewBox="0 0 11 10" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M10.6 0.200059C11.0418 0.53143 11.1314 1.15823 10.8 1.60006L4.8 9.60006C4.62607 9.83197 4.36005 9.97699 4.07089 9.99754C3.78173 10.0181 3.49788 9.91215 3.29289 9.70717L0.292893 6.70717C-0.0976311 6.31664 -0.0976311 5.68348 0.292893 5.29295C0.683417 4.90243 1.31658 4.90243 1.70711 5.29295L3.89181 7.47765L9.2 0.400059C9.53137 -0.0417689 10.1582 -0.131312 10.6 0.200059Z" fill="white"/></svg>\');'
        : null,
    backgroundColor: getBackgroundColor(props),
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    ':hover': activeStyle,
    ':active': activeStyle,
  };
});

export const Label = styled('div', props => {
  const {$theme} = props;
  const {typography} = $theme;
  return {
    verticalAlign: 'middle',
    ...getLabelPadding(props),
    color: getLabelColor(props),
    ...typography.font400,
  };
});
// tricky style for focus event cause display: none doesn't work
export const Input = styled('input', {
  opacity: 0,
  width: 0,
  overflow: 'hidden',
  margin: 0,
  padding: 0,
});
