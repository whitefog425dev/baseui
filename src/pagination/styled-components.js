/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import {styled} from '../styles/index.js';
import {StyledList} from '../menu/index.js';
import {StyledBaseButton} from '../button/index.js';

export const Root = styled('div', {
  display: 'flex',
  alignItems: 'center',
});
Root.displayName = 'StyledRoot';

export const MaxLabel = styled('span', ({$theme}) => ({
  ...$theme.typography.font300,
  marginLeft: $theme.sizing.scale300,
  marginRight: $theme.sizing.scale600,
}));
MaxLabel.displayName = 'StyledMaxLabel';

export const DropdownContainer = styled('div', ({$theme}) => ({
  position: 'relative',
  marginLeft: $theme.sizing.scale600,
  marginRight: $theme.sizing.scale300,
}));
DropdownContainer.displayName = 'StyledDropdownContainer';

export const DropdownMenu = styled(StyledList, ({$theme}) => ({
  position: 'absolute',
  overflow: 'auto',
  maxHeight: '200px',
  top: 'auto',
  marginTop: $theme.sizing.scale300,
  left: 0,
  right: 0,
}));
DropdownMenu.displayName = 'StyledDropdownMenu';

export const DropdownButton = styled(StyledBaseButton, ({$theme}) => ({
  color: $theme.colors.foreground,
  minWidth: `calc(${$theme.sizing.scale1600} + ${$theme.sizing.scale400})`,
}));
DropdownButton.displayName = 'StyledDropdownButton';
