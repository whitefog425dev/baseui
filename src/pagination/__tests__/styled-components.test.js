/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import * as React from 'react';
import {shallow} from 'enzyme';
import * as StyledComponents from '../styled-components.js';

function makeTest({
  title,
  component: Component,
  props = {},
  children = null,
  snapshotName = 'correct styles',
}: {
  title: string,
  component: React.ComponentType<*>,
  props?: {},
  children?: *,
  snapshotName?: string,
}) {
  test(title, () => {
    const shallowed = shallow(<Component {...props}>{children}</Component>);
    expect(shallowed.instance().getStyles()).toMatchSnapshot(snapshotName);
  });
}

describe('Pagination Styled Components', () => {
  Object.keys(StyledComponents).forEach(componentName => {
    makeTest({
      title: `${componentName} - basic renders`,
      component: StyledComponents[componentName],
    });
  });
});
