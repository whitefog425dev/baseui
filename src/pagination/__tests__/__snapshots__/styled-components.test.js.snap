// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Styled Components DropdownButton - basic renders: correct styles 1`] = `
Object {
  "color": "$theme.colors.black",
  "minWidth": "calc($theme.sizing.scale1600 + $theme.sizing.scale400)",
}
`;

exports[`Pagination Styled Components DropdownContainer - basic renders: correct styles 1`] = `
Object {
  "marginLeft": "$theme.sizing.scale600",
  "marginRight": "$theme.sizing.scale300",
  "position": "relative",
}
`;

exports[`Pagination Styled Components DropdownMenu - basic renders: correct styles 1`] = `
Object {
  "left": 0,
  "marginTop": "$theme.sizing.scale300",
  "maxHeight": "200px",
  "overflow": "auto",
  "position": "absolute",
  "right": 0,
  "top": "auto",
}
`;

exports[`Pagination Styled Components MaxLabel - basic renders: correct styles 1`] = `
Object {
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "marginLeft": "$theme.sizing.scale300",
  "marginRight": "$theme.sizing.scale600",
}
`;

exports[`Pagination Styled Components Root - basic renders: correct styles 1`] = `
Object {
  "alignItems": "center",
  "display": "flex",
}
`;
